test/integration/controller/job/job_webhook_test.go

```diff
@@ -154,4 +162,52 @@ var _ = ginkgo.Describe("Job Webhook", func() {
 			gomega.Expect(k8sClient.Update(ctx, createdJob)).ShouldNot(gomega.Succeed())
 		})
 	})
+
+	ginkgo.When("With manageJobsWithoutQueueName enabled and server version less than 1.27", ginkgo.Ordered, ginkgo.ContinueOnFailure, func() {
+		ginkgo.BeforeAll(func() {
+			fakeDiscoveryClient, _ := fakeclient.NewSimpleClientset().Discovery().(*fakediscovery.FakeDiscovery)
+			fakeDiscoveryClient.FakedServerVersion = &version.Info{GitVersion: "1.26.3"}
+			serverVersionFetcher := kubeversion.NewServerVersionFetcher(fakeDiscoveryClient)
```

Comment:
In an integration test, the idea is that we test against the real apiserver.
By faking the API, this test is not much more than a unit test, so I think we shouldn't need it.

However, it looks like through this test you discovered that the webhook was not properly configured. But in that case, we can have an integration test for any of the other validations that doesn't require us to fake any of the clients.

Reply:
Let's remove this test and add a separate test (without fakes) as part of #1035